eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,225 @@
+    private static final int[] TOKENS = new int[] {
+        TokenTypes.METHOD_CALL,
+        TokenTypes.VARIABLE_DEF
+    };
+
+    private static final Map<String, String> loggerMethodMap = new HashMap<>();
+
+    // Logger level static final string variables
+    private static final String FATAL = "fatal";
+    private static final String ERROR = "error";
+    private static final String WARN = "warn";
+    private static final String INFO = "info";
+    private static final String DEBUG = "debug";
+    private static final String TRACE = "trace";
+
+    private static final String IS_FATAL_ENABLED = "isFatalEnabled";
+    private static final String IS_ERROR_ENABLED = "isErrorEnabled";
+    private static final String IS_WARN_ENABLED = "isWarnEnabled";
+    private static final String IS_INFO_ENABLED = "isInfoEnabled";
+    private static final String IS_DEBUG_ENABLED = "isDebugEnabled";
+    private static final String IS_TRACE_ENABLED = "isTraceEnabled";
+
+    private static final Set<String> loggerSet = new HashSet<>();
+
+    @Override
+    public void init() {
+        initLogLevelMap();
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return TOKENS;
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+
+        // skip check if no Logger class imported
+        if (!hasLoggerImport(ast)) {
+            return;
+        }
+
+        switch (ast.getType()) {
+            case TokenTypes.VARIABLE_DEF:
+                isNameLogger(ast);
+                isNonStaticLoggerInstance(ast);
+                break;
+            case TokenTypes.METHOD_CALL:
+                isGuardConditionChecked(ast);
+                break;
+        }
+    }
+
+    /**
+     * Check if the file has Logger class imported
+     * @param rootAST The root of AST
+     * @return true if the class has Logger class imported.
+     */
+    private boolean hasLoggerImport(DetailAST rootAST) {
+        DetailAST importToken = rootAST.findFirstToken(TokenTypes.IMPORT);
+        while (importToken != null && importToken.getType() == TokenTypes.IMPORT ) {
+            String importClassName = (importToken.findFirstToken(TokenTypes.DOT)).getLastChild().getText();
+            if (importClassName.equals(LOGGER_CLASS)) {
+                return true;
+            }
+            importToken = importToken.getNextSibling();
+        }
+        return false;
+    }
+
+    /**
+     * Check if the Logger instance named as logger, log as error if not
+     * @param varDefAST the VARIABLE_DEF node
+     */
+    private void isNameLogger(DetailAST varDefAST) {
+        String type = varDefAST.findFirstToken(TokenTypes.TYPE).getFirstChild().getText();
+        if (type.equals(LOGGER_CLASS)) {
```

Comment:
Using TokenUtil would make this easier. http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/utils/TokenUtil.html

Reply:
In general, you can replace some of these with TokenUtil.findEachTokenWithPredicate